code/game/gamemodes/clock_cult/clock_objects.dm

```diff
@@ -0,0 +1,1568 @@
+			C << "<span class='warning'>You require a free hand to utilize [src]'s power!</span>"
+			return 0
+		C.visible_message("<span class='warning'>[C]'s hand is enveloped in violet flames!<span>", "<span class='heavy_brass'>You harness [src]'s power. Direct it at a tile on harm intent to unleash it, or hit your visor with it dispel it.</span>")
+		var/obj/item/weapon/ratvars_flame/R = new(get_turf(C))
+		flame = R
+		C.put_in_hands(R)
+		R.visor = src
+		return 1
+
+/obj/item/clothing/glasses/judicial_visor/proc/update_status(change_to)
+	if(recharging)
+		icon_state = "judicial_visor_0"
+		return 0
+	if(active == change_to)
+		return 0
+	if(!isliving(loc))
+		return 0
+	var/mob/living/L = loc
+	if(!is_servant_of_ratvar(L) || L.stat)
+		return 0
+	active = change_to
+	icon_state = "judicial_visor_[active]"
+	switch(active)
+		if(TRUE)
+			L << "<span class='notice'>As you put on [src], its lens begins to glow, information flashing before your eyes.</span>\n\
+			<span class='heavy_brass'>Judicial visor active. Alt-click visor to gain the ability to smite the unworthy.</span>"
+		if(FALSE)
+			L << "<span class='notice'>As you take off [src]. its lens darkens once more.</span>"
+	return 1
+
+/obj/item/clothing/head/helmet/clockwork //Clockwork armor: High melee protection but weak to lasers
+	name = "clockwork helmet"
+	desc = "A heavy helmet made of brass."
+	icon = 'icons/obj/clothing/clockwork_garb.dmi'
+	icon_state = "clockwork_helmet"
+	w_class = 3
+	armor = list(melee = 65, bullet = 50, laser = -25, energy = 5, bomb = 0, bio = 0, rad = 0)
+
+/obj/item/clothing/suit/armor/clockwork
+	name = "clockwork cuirass"
+	desc = "A bulky cuirass made of brass."
+	icon = 'icons/obj/clothing/clockwork_garb.dmi'
+	icon_state = "clockwork_cuirass"
+	w_class = 4
+	armor = list(melee = 80, bullet = 40, laser = -10, energy = 5, bomb = 0, bio = 0, rad = 0)
+	allowed = list(/obj/item/clockwork)
+
+/obj/item/clothing/shoes/clockwork
+	name = "clockwork treads"
+	desc = "Industrial boots made of brass. They're very heavy."
+	icon = 'icons/obj/clothing/clockwork_garb.dmi'
+	icon_state = "clockwork_treads"
+	w_class = 3
+	flags = NOSLIP
+
+/obj/item/weapon/ratvars_flame //Used by the judicial visor
+	name = "Ratvar's flame"
+	desc = "A blazing violet ball of fire that, curiously, doesn't melt your hand off."
+	icon = 'icons/effects/clockwork_effects.dmi'
+	icon_state = "ratvars_flame" //beware: minecraft-tier sprite
+	item_state = "disintegrate"
+	color = rgb(180, 50, 210)
+	w_class = 5
+	flags = NODROP | ABSTRACT
+	force = 15 //Also serves as a potent melee weapon!
+	damtype = BURN
+	hitsound = 'sound/weapons/sear.ogg'
+	attack_verb = list("scorched")
+	var/obj/item/clothing/glasses/judicial_visor/visor //The linked visor
+
+/obj/item/weapon/ratvars_flame/afterattack(atom/target, mob/living/user, flag, params)
+	if(!visor || (visor && visor.cooldown))
+		qdel(src)
+	if(user.a_intent != "harm")
+		..()
+		return 0
+	visor.recharging = TRUE
```

Comment:
This is bad.

You should use world.time timestamps to see if it'd ready again instead.